apps/rehabstod/src/pages/Patient/components/SelectRekoStatus/SelectRekoStatus.tsx

```diff
@@ -1,42 +1,44 @@
-import { useState } from 'react'
-import { useDispatch } from 'react-redux'
+import { useEffect, useState } from 'react'
 import { SelectFilter } from '../../../../components/Table/filter/SelectFilter'
-import { RekoStatus, RekoStatusType } from '../../../../schemas/sickLeaveSchema'
+import { RekoStatusType } from '../../../../schemas/sickLeaveSchema'
 import { useAppSelector } from '../../../../store/hooks'
-import { useSetRekoStatusMutation } from '../../../../store/sickLeaveApi'
+import { useGetRekoStatusForPatientQuery, useSetRekoStatusMutation } from '../../../../store/sickLeaveApi'
 import { getRekoStatusSickLeaveTimestamp } from '../../../../utils/getRekoStatusSickLeaveTimestamp'
-import { updateRekoStatusId } from '../../../../store/slices/sickLeave.slice'
 
 export function SelectRekoStatus({
   patientId,
   endDate,
+  startDate,
   rekoStatusTypes,
   description,
   disabled = false,
 }: {
   patientId: string
   endDate: string
+  startDate: string
   rekoStatusTypes: RekoStatusType[]
   description: string
   disabled?: boolean
 }) {
   const [setRekoStatus] = useSetRekoStatusMutation()
-  const { rekoStatusId } = useAppSelector((rootState) => rootState.sickLeave)
+  const { data: rekoStatusFromDatabase } = useGetRekoStatusForPatientQuery({ patientId, endDate, startDate })
   const emptyRekoStatus = rekoStatusTypes.find((rekoStatusType) => rekoStatusType.name === 'Ingen')
-  const [savedRekoStatus, updateSavedRekoStatus] = useState(rekoStatusId || emptyRekoStatus?.id)
+  const [savedRekoStatus, updateSavedRekoStatus] = useState(rekoStatusFromDatabase?.status.id || emptyRekoStatus?.id)
   const sickLeaveTimestamp = getRekoStatusSickLeaveTimestamp(endDate)
   const { filter } = useAppSelector((state) => state.sickLeave)
-  const dispatch = useDispatch()
 
   const handleSetRekoStatus = (id: string) => {
     const type = rekoStatusTypes.find((rekoStatusType) => id === rekoStatusType.id)
     if (type) {
       setRekoStatus({ patientId, status: type, sickLeaveTimestamp, filter })
       updateSavedRekoStatus(type.id)
-      dispatch(updateRekoStatusId(type.id))
     }
   }
 
+  useEffect(() => {
+    updateSavedRekoStatus(rekoStatusFromDatabase?.status.id)
+  }, [rekoStatusFromDatabase])
+
```

Comment:
Not sure if we need to store `rekostatus` in a local state. We could use the state that the API provides.

```typescript
const savedRekoStatus = rekoStatusFromDatabase?.status.id || emptyRekoStatus?.id
```